airflow/providers/google/cloud/transfers/gcs_to_bigquery.py

```diff
@@ -229,105 +259,307 @@ def __init__(
+                        f'want to force rerun it consider setting `force_rerun=True`.'
+                        f'Or, if you want to reattach in this scenario add {job.state} to `reattach_states`'
+                    )
+
+            job_types = {
+                LoadJob._JOB_TYPE: ["sourceTable", "destinationTable"],
+                CopyJob._JOB_TYPE: ["sourceTable", "destinationTable"],
+                ExtractJob._JOB_TYPE: ["sourceTable"],
+                QueryJob._JOB_TYPE: ["destinationTable"],
+            }
+
+            if self.hook.project_id:
+                for job_type, tables_prop in job_types.items():
+                    job_configuration = job.to_api_repr()["configuration"]
+                    if job_type in job_configuration:
+                        for table_prop in tables_prop:
+                            if table_prop in job_configuration[job_type]:
+                                table = job_configuration[job_type][table_prop]
+                                persist_kwargs = {
+                                    "context": context,
+                                    "task_instance": self,
+                                    "project_id": self.hook.project_id,
+                                    "table_id": table,
+                                }
+                                if not isinstance(table, str):
+                                    persist_kwargs["table_id"] = table["tableId"]
+                                    persist_kwargs["dataset_id"] = table["datasetId"]
+                                BigQueryTableLink.persist(**persist_kwargs)
+
+            self.job_id = job.job_id
+            context["ti"].xcom_push(key="job_id", value=self.job_id)
+            if self.deferrable:
+                self.defer(
+                    timeout=self.execution_timeout,
+                    trigger=BigQueryInsertJobTrigger(
+                        conn_id=self.gcp_conn_id,
+                        job_id=self.job_id,
+                        project_id=self.hook.project_id,
+                    ),
+                    method_name="execute_complete",
+                )
+            else:
+                job.result(timeout=self.result_timeout, retry=self.result_retry)
+                max_id = self._find_max_value_in_column()
+                self._handle_job_error(job)
+                return max_id
+
+    def execute_complete(self, context: Context, event: dict[str, Any]):
+        """
+        Callback for when the trigger fires - returns immediately.
+        Relies on trigger to throw an exception, otherwise it assumes execution was
+        successful.
+        """
+        if event["status"] == "error":
+            raise AirflowException(event["message"])
+        self.log.info(
+            "%s completed with response %s ",
+            self.task_id,
+            event["message"],
+        )
+        self._find_max_value_in_column()
+        return self.job_id
```

Comment:
if the normal version of the operators returns max_id I think this method should return the same value

Reply:
Done